Remove dependency on Schema v1.1

nigelmegitt opened the pull request:
Also make the metadata datatypes in a different namespace from the normal datatypes, for clash avoidance in downstream (referring) schemas.

nigelmegitt committed:
Remove dependency on Schema v1.1

Also make the metadata datatypes in a different namespace from the normal datatypes, for clash avoidance in downstream (referring) schemas.

andreastai commented:
@nigelmegitt Thanks for the adjustments!

The following construct seems to lead to false-negative validations messages:

```xml
	<xs:complexType name="headMetadata_type">
....
					<xs:any minOccurs="0" maxOccurs="unbounded"
						processContents="lax" namespace="##local"
					/>
				</xs:sequence>
....
		</xs:complexContent>
	</xs:complexType>
```
By setting namespace to `##local` only elements that are not namespace qualified are allowed.

This makes 

```xml
<tt:head>
  <tt:metadata>
    <ebuttm:foo>bar</ebuttm:foo>
  </tt:metadata>
</tt:head>
```
invalid but 


```xml
<tt:head>
  <tt:metadata>
    <foo>bar</foo>
 </tt:metadata>
</tt:head>
```
valid.

andreastai reviewed (changes requested):
There seems to be an error. 

@nigelmegitt It would be great if you could double-check it (see comment). Thank you.

andreastai commented:
As it seems to fix this issue it just needed to remove the `namespace` attribute.  

@nigelmegitt I am happy to provide a pull request and continue the review.

nigelmegitt commented:
I've had a look at this in some more detail and posted my findings at https://github.com/ebu/ebu-tt-m-xsd/issues/30#issuecomment-564622613 - in short, we can't remove the `namespace` attribute but we could remove almost all the detailed per-parent-element metadata validation in favour of a single `metadata` element definition that allows anything in any namespace other than the TT namespace, using `namespace="##other" processContents="lax"`. If we were to do that, then undefined elements would not be validated other than for XML parsing validity, _even if they are in known namespaces_.

nigelmegitt committed:
Address review comment

Removes detailed metadata types and replaces with a single generic one. Declares the `documentMetadata` element so that it may be used within the `metadata` element.

nigelmegitt commented:
@tairt Please check 4026249 does what you expect. In dependent repos I will open pull requests to refer to the newly defined `ebuttm:metadata_type` for the `tt:metadata` element instead of the separate types for head, body etc.

andreastai commented:
https://github.com/ebu/ebu-tt-m-xsd/commit/4026249a949f594ae0d38b84e22cf2e8e665f235 does not solve the issue. The value for the `@namespace` attribute is `##other`. This allows any namespace except elements within the target namespace. The type is defined in a schema document with the target namespace `urn:ebu:tt:metadata`. This means that any EBU-TT Metadata vocabulary would not be allowed.

andreastai commented:
To get the desired functionality, the type needs to be defined in the schema document where the elements of the namespace `http://www.w3.org/ns/ttml` are defined (e.g. in the EBU-TT Part 1 schema it would be `ebutt.xsd`).
